client/src/Pages/TripDetails.tsx

```diff
@@ -20,13 +21,15 @@ type TripProps = {
 export default function TripDetails({ onClick }: TripProps) {
   const [activeIcon, setActiveIcon] = useState('');
   const [err, setError] = useState<Error>();
+  const [activeEventId, setActiveEventId] = useState<number | undefined>(1);
   const { tripId } = useParams();
   const navigate = useNavigate();
   const { trip, error, isLoading } = useFindTrip(1, Number(tripId));
 
   useEffect(() => {
-    if (trip) setActiveIcon(trip.iconUrl);
+    if (trip) setActiveIcon(trip[0].iconUrl);
```

Comment:
Is this safe? Are you guaranteed that `trip` will have at least one element in it? Maybe you should say
`if (trip?.[0]) setActiveIcon(trip[0].iconUrl);`